### ISSUE TITLE
Total loss is NaN while other losses are not

### ISSUE THREAD

**Issue by zb1439** [matterport/Mask_RCNN#1780]
I train a new model based on your implementation and the compile method is not even changed.
I am sure that my training labels did not exceed the number of categories. I note that even when the very step that the total loss gives nan, the other loss seems to be normal. 

Epoch 2/40

 610/1000 [=================>............] - ETA: 9:38 - loss: nan - rpn_class_loss: 0.4507 - rpn_bbox_loss: 0.3960 - mrcnn_class_loss: 0.0864 - mrcnn_bbox_loss: 0.2338 - mrcnn_mask_loss: 0.4258

**Comment by zb1439** [comment]
I just figured out where the nan comes from. It was the L2-REGULARIZATION that suddenly turns nan during training. I used 0.0001 as weight decay, which I thought would be small enough. 
I am trying fixing this with one line of code in the compile.

`reg_loss = K.switch(tf.is_nan(reg_loss), tf.constant(0.0), reg_loss)`

I will try lower my weight decay and see if these two methods gives any significant impact to the model. But I'm really curious why it happens with such a generally suitable weight decay.

**Comment by zb1439** [comment]
This time, however, none of the loss turns nan, but the model is still strange as suddenly all losses come up and the phenomenon is almost the same as before, when loss turns nan.
